splash/qtrender_lua.py

```diff
@@ -487,6 +491,76 @@ def error(error_info):
         ))
         return cmd
 
+    @command(async=True, sets_callback=True, decode_arguments=False)
+    def with_timeout(self, callback, timeout):
+        if timeout is None:
+            ScriptError({
+                "argument": "timeout",
+                "message": "splash:with_timeout timeout must be specified",
+                "splash_method": "with_timeout",
+            })
+        if not isinstance(timeout, (float, int)):
+            raise ScriptError({
+                "argument": "timeout",
+                "message": "splash:with_timeout timeout must be a number",
+                "splash_method": "with_timeout",
+            })
+
+        timeout = int(float(timeout) * 1000)
+
+        if timeout < 0:
+            raise ScriptError({
+                "argument": "timeout",
+                "message": "splash:with_timeout timeout must be >= 0",
+                "splash_method": "with_timeout",
+            })
+
+        if lupa.lua_type(callback) != 'function':
+            raise ScriptError({
+                "argument": "callback",
+                "message": "splash:with_timeout callback is not a function",
+                "splash_method": "with_timeout",
+            })
+
+        qtimer = QTimer(self.tab)
+        qtimer.setSingleShot(True)
+
+        def timer_callback():
+            run_coro.runner.stop()
+            cmd.return_result(None, "timeout_over")
+
+        qtimer.timeout.connect(timer_callback)
+
+        def coro_success(result):
+            if not qtimer.isActive():
+                return
+
+            qtimer.stop()
+            cmd.return_result(True, result)
+
+        def coro_error(ex):
+            if not qtimer.isActive():
+                return
+
+            qtimer.stop()
+
+            info = str(ex.args[0]["error"])
+
+            cmd.return_result(None, info)
```

Comment:
a nitpick: could you please remove blank lines between two previous commands? PEP8 [says](https://www.python.org/dev/peps/pep-0008/#blank-lines) "Use blank lines in functions, *sparingly*, to indicate logical sections. "